### PR TITLE
Fix bug where redeclaration of var doesn't deopt

### PR DESCRIPTION
**Pull request by boopathi**
https://phabricator.babeljs.io/T7470

### CODE REVIEW

File: packages/babel-traverse/test/evaluation.js [hunk 1]
```diff
@@ -37,4 +37,11 @@ suite("evaluation", function () {
       false
     );
   });
+
+  test("should not be confident when var is redeclared in the same scope", function () {
+    assert.strictEqual(
+      getPath("var x = 2; var y = x + 2; { var x = 3 }").get("body.1.declarations.0.init").evaluate().confident,
```
Comment (hzoo) [inline packages/babel-traverse/test/evaluation.js]: Looks good to fix the bug, but as another TODO `y` be able to resolve to `4` right?

Comment (boopathi) [inline packages/babel-traverse/test/evaluation.js]: Do you mean `var x = 2; var y = x + 2` with y resolved to 4 ?

**Comment by boopathi** [comment]
Added a few other tests.

File: packages/babel-traverse/test/evaluation.js (hunk 1 above)
Comment (hzoo) [inline packages/babel-traverse/test/evaluation.js]: Yeah `var x = 2; var y = 4; { var x = 3; }`

Comment (boopathi) [inline packages/babel-traverse/test/evaluation.js]: It deopts when there is one more assignment in any of the branches. So I don't think it'll become `y=4`.

Comment (boopathi) [inline packages/babel-traverse/test/evaluation.js]: http://astexplorer.net/#/XYjKjnPM46/1

Comment (hzoo) [inline packages/babel-traverse/test/evaluation.js]: Oh yeah I just meant evaluate could be smarter to do so (didn't mean this PR)

**Comment by hzoo** [comment]
:+1: @loganfsmyth

**Comment by boopathi** [comment]
Rebased with master and fixed conflicts.

**Pull request merged**